packages/karma-runner/src/karma-plugins/TestHooksMiddleware.ts

```diff
@@ -0,0 +1,111 @@
+import * as path from 'path';
+import * as url from 'url';
+
+import { RequestHandler } from 'express';
+import { CoverageAnalysis } from '@stryker-mutator/api/core';
+import { MutantRunOptions } from '@stryker-mutator/api/test_runner2';
+import { escapeRegExp } from '@stryker-mutator/util';
+
+export const TEST_HOOKS_FILE_NAME = __filename;
+
+const SUPPORTED_FRAMEWORKS = Object.freeze(['mocha', 'jasmine'] as const);
+
+type SupportedFramework = 'mocha' | 'jasmine';
+
+function isSupportedFramework(framework: string): framework is SupportedFramework {
+  return (SUPPORTED_FRAMEWORKS as readonly string[]).includes(framework);
+}
+
+/**
+ * Keep in sync with StrykerMutantCoverageAdapter.ts
+ */
+const SHOULD_REPORT_COVERAGE_FLAG = '__strykerShouldReportCoverage__';
+
+export default class TestHooksMiddleware {
+  private static _instance: TestHooksMiddleware;
+  private testFramework: 'mocha' | 'jasmine' | undefined;
+  public currentTestHooks = '';
+
+  public static get instance(): TestHooksMiddleware {
+    if (!this._instance) {
+      this._instance = new TestHooksMiddleware();
+    }
+    return this._instance;
+  }
+
+  public configureTestFramework(frameworks?: string[]) {
```

Comment:
Why is this an array of frameworks and why is it optional?

Reply:
We're essentially passing the `frameworks` from karma options. It is in theory optional since you could choose to roll-your-own test framework. 

I wanted to pass them all, since I want all test framework specific stuff to be in the test hooks middleware.